### PR TITLE
Sample Code JSON metadata files

### PR DESCRIPTION
**Pull request by ChrisFrenchPDX**
PR to start the review process of the new JSON metadata workflow for the Developer site and Code Sample viewer.

### CODE REVIEW

**Comment by tdunn** [comment]
@ChrisFrenchPDX @mikewilburn - I will coordinate this review process for Qt.

**Comment by tdunn** [comment]
I'm about halfway done reviewing and updating the C++ samples' metadata.

**Comment by markjdugger** [comment]
@tdunn @ThadT Could one of your review commit 3b32b55de602cd173a184941157f2761785d8543. It contains all the QML files.

@ChrisFrenchPDX Once my commit has been reviewed by Tom or Thad, you can run your script against the README.metadata.json files and cross your fingers!

**Comment by tdunn** [comment]
@markjdugger - Only saw one problem, an empty-string as keyword at https://github.com/Esri/arcgis-runtime-samples-qt/blob/3b32b55de602cd173a184941157f2761785d8543/ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/Picture_Marker_Symbol/README.metadata.json#L9. 

Otherwise, it looks good. Thanks for cranking through that work.

**Comment by markjdugger** [comment]
@ChrisFrenchPDX The JSON metadata files are ready for you to test out your spiffy script.  Crossing my fingers!

File: ArcGISRuntimeSDKQt_QMLSamples/Maps/MinMaxScale/README.metadata.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        "screenshot.png"
+    ],
+    "keywords": [
+        "extent",
+        "maps",
+        "2D",
+        "scale",
+        "setminScale",
+        "setmaxScale",
+        "zoom",
```
Comment (ChrisFrenchPDX) [inline ArcGISRuntimeSDKQt_QMLSamples/Maps/MinMaxScale/README.metadata.json]:
```suggestion
        "zoom"
```

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/CreateTerrainSurfaceFromLocalTilePackage/README.metadata.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "category": "Scenes",
+    "description": "Set the terrain surface with elevation described by a local tile package",
+    "ignore": false,
+    "images": [
+        "screenshot.png"
+    ],
+    "keywords": [
+        "3D",
+        "tile cache",
+        "elevation surface",
```
Comment (ChrisFrenchPDX) [inline ArcGISRuntimeSDKQt_QMLSamples/Scenes/CreateTerrainSurfaceFromLocalTilePackage/README.metadata.json]:
```suggestion
        "elevation surface"
```

File: ArcGISRuntimeSDKQt_QMLSamples/Maps/MinMaxScale/README.metadata.json (hunk 1 above)
Comment (markjdugger) [reply ArcGISRuntimeSDKQt_QMLSamples/Maps/MinMaxScale/README.metadata.json]: In JSON, trailing comma illegal. Got it

File: ArcGISRuntimeSDKQt_QMLSamples/Scenes/CreateTerrainSurfaceFromLocalTilePackage/README.metadata.json (hunk 2 above)
Comment (markjdugger) [reply ArcGISRuntimeSDKQt_QMLSamples/Scenes/CreateTerrainSurfaceFromLocalTilePackage/README.metadata.json]: Agreed

**Comment by ChrisFrenchPDX** [comment]
@markjdugger @ThadT @tdunn Thanks for getting this done so quickly! Hopefully, I'll have a working PR up today or tomorrow, and then I'll provide the details about how to set it up on your local machine. 

@markjdugger Yes, the JSON library we're using panicked when it saw the trailing comma. I'm going to talk to @patrickarlt about what should happen if there are syntactic errors with the readme file: the questions is whether, when an io exception gets thrown, the readme should be skipped and any io exceptions logged, or should the Acetate build fail with an error message.

File: ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/Simple_Renderer/README.metadata.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
+    "category": "Display Information",
```
Comment (ChrisFrenchPDX) [inline ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/Simple_Renderer/README.metadata.json]:
```suggestion
    "category": "Display information",
```

The punctuation of category names matters for the nav bar: 

image

**Comment by mikewilburn** [comment]
@ChrisFrenchPDX seems like logging the IO exceptions (so that they can be addressed later), but still allowing the site to build, would be the way to go. If there are a handful of syntactic errors that the Acetate build script runs across, it would be tedious to address them (and have PRs reviewed/approved/merged) in time to ensure we can get a new site release out.

File: ArcGISRuntimeSDKQt_QMLSamples/EditData/DeleteFeaturesFeatureService/README.metadata.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        "SpatialReference",
+        "ServiceFeatureTable",
+        "FeatureLayer",
+        "Feature",
+        "ArcGISFeature",
+        "FeatureEditResult",
+        "FeatureQueryResult"
+    ],
+    "snippets": [
+        "PLEASE VERIFY THE INCLUSION OF THESE SNIPPETS ***",
```
Comment (ChrisFrenchPDX) [inline ArcGISRuntimeSDKQt_QMLSamples/EditData/DeleteFeaturesFeatureService/README.metadata.json]:
```suggestion
```

**Comment by patrickarlt** [comment]
@markjdugger @tdunn @mikewilburn @ChrisFrenchPDX RE: https://github.com/Esri/arcgis-runtime-samples-qt/pull/889#issuecomment-492809873

The main issue we don't have a ways to capture these errors and distribute them to teams. The best we can do is look at the build log and manually check for any sort of warning that we output. If we don't error builds it is REALLY probable that samples or snippets will go missing without a regular audit.

I'm fine that but it will be the samples teams responsibility to do the audit following directions that we (dev site team) will write up once we release this.

**Comment by patrickarlt** [comment]
@mikewilburn @tdunn @markjdugger We are ready on our side for the inclusion of this PR. We need this to merge today or early tomorrow to finish prepping for release on our end which includes:

1. Remove qt samples from the `//core` whitelist
2. Delete QT samples from `//core`
3. Verify changes
4. Change sample layout to include new mega menus if that merges this week https://github.com/ArcGIS/arcgis-for-developers/pull/8374.

**Comment by tdunn** [comment]
@patrickarlt @ChrisFrenchPDX - Sounds great! Looking forward to testing the final product. Thanks.

**Pull request merged**